python/ray/experimental/data/dataset.py

```diff
@@ -13,7 +13,11 @@
     import pyspark
     import ray.util.sgd
     import torch
+
+try:
     import tensorflow as tf
+except ImportError:
+    tf = None
 
```

Comment:
Could we defer this import until the to tf call? You can import it under the type checking block still.